src/microbe_stage/MicrobeInternalCalculations.cs

```diff
@@ -1,39 +1,147 @@
-                    * directionFactor;
+                // We get the movement force for every direction as well
+                forwardsDirectionMovementForce += MovementForce(movementConstant, forwardDirectionFactor);
+                backwardsDirectionMovementForce += MovementForce(movementConstant, backwardDirectionFactor);
+                rightwardDirectionMovementForce += MovementForce(movementConstant, rightwardDirectionFactor);
+                leftwardDirectionMovementForce += MovementForce(movementConstant, leftwardDirectionFactor);
             }
         }
 
+        var maximumMovementDirection = MaximumSpeedDirection(organelles);
+
+        // After getting the maximum-force direction we normalize it
+        maximumMovementDirection = maximumMovementDirection.Normalized();
+
+        // If the flagella are positioned symetrically we assume the forward position as default
+        if (maximumMovementDirection == Vector3.Zero)
+            maximumMovementDirection = Vector3.Forward;
+
+        // Calculate the maximum total force-factors in the maximum-force direction
+        forwardDirectionFactor = maximumMovementDirection.Dot(Vector3.Forward);
+        backwardDirectionFactor = -forwardDirectionFactor;
+        rightwardDirectionFactor = maximumMovementDirection.Dot(Vector3.Right);
+        leftwardDirectionFactor = -rightwardDirectionFactor;
+
+        // Add each movement force to the total movement force in the maximum-force direction.
+        organelleMovementForce += MovementForce(forwardsDirectionMovementForce, forwardDirectionFactor);
+        organelleMovementForce += MovementForce(backwardsDirectionMovementForce, backwardDirectionFactor);
+        organelleMovementForce += MovementForce(rightwardDirectionMovementForce, rightwardDirectionFactor);
+        organelleMovementForce += MovementForce(leftwardDirectionMovementForce, leftwardDirectionFactor);
+
         float baseMovementForce = Constants.CELL_BASE_THRUST *
             (membraneType.MovementFactor - membraneRigidity * Constants.MEMBRANE_RIGIDITY_MOBILITY_MODIFIER);
 
         float finalSpeed = (baseMovementForce + organelleMovementForce) / microbeMass;
 
         return finalSpeed;
     }
+
+    private static float MovementForce(float movementForce, float directionFactor)
+    {
+        if (directionFactor < 0)
+            return 0;
+
+        return movementForce * directionFactor;
+    }
+
+    // Symetric flagella are a corner case for speed calculations because the sum of all
+    // directions is kinda of broken in their case, so we have to choose which one of the symmetric flagella
+    // we must discard from the direction calculation
+    // Here we only discared if the flagella we input is the "bad" one
+    private static Vector3 ChooseFromSymetricFlagella(IEnumerable<OrganelleTemplate> inputOrganelles,
+        OrganelleTemplate testedOrganelle, Vector3 maximumMovementDirection)
+    {
+        var organelles = inputOrganelles.ToList();
```

Comment:
This ToList call is completely unnecessary

Reply:
jetbrains gives me an error for multiple enumerations if i iterate multiple times through it.

Reply:
But you do not? I don't see the organelles variable being used anywhere else in this method than on the next line.

Reply:
yea, i didint see that there